packages/@lwc/ssr-runtime/src/reflection.ts

```diff
@@ -16,16 +17,85 @@ import {
 
 import type { LightningElement } from './lightning-element';
 
-// Eventually include globals that also reflect
-const attrsToProps = AriaAttrNameToPropNameMap;
+/**
+ * Map of global attribute or ARIA attribute to the corresponding property name.
+ * Not all global attributes are included, just those from `HTMLElementTheGoodParts`.
+ */
+const attrsToProps = assign(create(null), {
+    accesskey: 'accessKey',
+    dir: 'dir',
+    draggable: 'draggable',
+    hidden: 'hidden',
+    id: 'id',
+    lang: 'lang',
+    spellcheck: 'spellcheck',
+    tabindex: 'tabIndex',
+    title: 'title',
+    ...AriaAttrNameToPropNameMap,
+});
+
+const stringDescriptor = (attrName: string): TypedPropertyDescriptor<string | null> => ({
+    configurable: true,
+    enumerable: true,
+    get(this: LightningElement): string | null {
+        return this.getAttribute(attrName);
+    },
+    set(this: LightningElement, newValue: string | null): void {
+        const currentValue = this.getAttribute(attrName);
+        const normalizedValue = String(newValue);
+        if (normalizedValue !== currentValue) {
+            this.setAttribute(attrName, normalizedValue);
+        }
+    },
+});
+
+const booleanDescriptor = (
+    attrName: string,
+    defaultValue: boolean
+): TypedPropertyDescriptor<boolean> => ({
+    configurable: true,
+    enumerable: true,
+    get(this: LightningElement): boolean {
+        const value = this.getAttribute(attrName);
+        return value === null ? defaultValue : value === String(defaultValue);
+    },
+    set(this: LightningElement, newValue: boolean): void {
+        const currentValue = this.getAttribute(attrName);
+        const normalizedValue = String(Boolean(newValue));
+        if (normalizedValue !== currentValue) {
+            this.setAttribute(attrName, normalizedValue);
+        }
+    },
+});
```

Comment:
From some quick testing in Chrome:
- `draggable` never returns `true` unless its attribute is set to `"true"`
  - `$0.setAttribute('draggable', '') => false`
  - `$0.setAttribute('draggable', 'true') => true`
  - `$0.setAttribute('draggable', 'false') => false`
  - `$0.setAttribute('draggable', 'foo') => false`
  - `$0.draggable = null | undefined` normalizes reflection to `draggable="false"`
  - default value is `false`

- `hidden` behaves sanely; it returns `true` if the attribute is rendered, regardless of value
  - `$0.setAttribute('hidden', '') => true`
  - `$0.setAttribute('hidden', 'true') => true`
  - `$0.setAttribute('hidden', 'false') => true`
  - `$0.setAttribute('hidden', 'foo') => true`
  - `el.hidden = null | undefined` removes the attribute
  - default value is `false`

- `spellcheck` never returns `false` unless the attribute is set to `"false"`
  - `$0.setAttribute('spellcheck', '') => true`
  - `$0.setAttribute('spellcheck', 'true') => true`
  - `$0.setAttribute('spellcheck', 'false') => false`
  - `$0.setAttribute('spellcheck', 'foo') => true`
  - `$0.spellcheck = null | undefined` normalizes reflection to `spellcheck="false"`
  - default value is `true`

Reply:
Last I checked, these also vary across browsers:

https://github.com/salesforce/lwc/blob/75ff766c8dabdd28e3308e3f2c8f763c385b056a/packages/%40lwc/integration-karma/test/component/html-properties/index.spec.js#L133-L136

https://github.com/salesforce/lwc/blob/75ff766c8dabdd28e3308e3f2c8f763c385b056a/packages/%40lwc/integration-karma/test/component/html-properties/index.spec.js#L147-L151

Reply:
This helper implements the behavior for `draggable` / `spellcheck` (unless I did it wrong). I missed that `hidden` has different behavior. I'll fix that.

How do we want to handle the differences in default values? I'm inclined to just go with Chrome's default for now, and then do a follow up PR to conditionally ignore hydration errors on the relevant attributes.

Reply:
> Last I checked, these also vary across browsers:
> 
> https://github.com/salesforce/lwc/blob/75ff766c8dabdd28e3308e3f2c8f763c385b056a/packages/%40lwc/integration-karma/test/component/html-properties/index.spec.js#L133-L136
> 
> https://github.com/salesforce/lwc/blob/75ff766c8dabdd28e3308e3f2c8f763c385b056a/packages/%40lwc/integration-karma/test/component/html-properties/index.spec.js#L147-L151

I just checked, `tabIndex`/`role`/`draggable`/`hidden` have consistent default values in latest Chrome/Firefox/Safari for `document.createElement('div')`.